### PR TITLE
Refactor GPU examples tests

### PR DESCRIPTION
**Pull request by carmocca**
## What does this PR do?

Simplifying our GPU CI. No workload added or removed

## Before submitting
- [ ] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [n/a] Did you make sure to update the documentation with your changes? (if necessary)
- [n/a] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [n/a] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

### CODE REVIEW

File: .azure-pipelines/gpu-tests.yml [hunk 1]
```diff
@@ -104,10 +104,10 @@ jobs:
...
-        set -e
         python -m pytest pl_examples -v --maxfail=2 --durations=0
-        bash pl_examples/run_examples-args.sh --trainer.gpus 1 --trainer.max_epochs 1 --data.batch_size 64 --trainer.limit_train_batches 5 --trainer.limit_val_batches 3
-        bash pl_examples/run_ddp-examples.sh --trainer.max_epochs 1 --data.batch_size 32 --trainer.limit_train_batches 2 --trainer.limit_val_batches 2
+        bash pl_examples/run_examples.sh --trainer.gpus=1
+        bash pl_examples/run_examples.sh --trainer.gpus=2 --trainer.accelerator=ddp
+        bash pl_examples/run_examples.sh --trainer.gpus=2 --trainer.accelerator=ddp --trainer.precision=16
       env:
         PL_USE_MOCKED_MNIST: "1"
       displayName: 'Examples'
```
Comment (carmocca) [inline .azure-pipelines/gpu-tests.yml:113]:
```suggestion
      displayName: 'Testing: examples'
```

File: pl_examples/run_examples-args.sh [hunk 2]
```diff
@@ -1,15 +0,0 @@
-#!/bin/bash
-
-echo $@
-
-full_path=$(realpath $0)
-echo $full_path
-
-dir_path=$(dirname $full_path)
```
Comment (Borda) [inline pl_examples/run_examples-args.sh:8]: If you drop this, you cannot call it from elsewhere and we need it for ngc testing

**Review by Borda — CHANGES_REQUESTED** [review]

File: pl_examples/run_examples-args.sh (hunk 2 above)
Comment (carmocca) [reply pl_examples/run_examples-args.sh:8]: Updated

**Review by awaelchli — APPROVED** [review]

File: pl_examples/run_examples-args.sh (hunk 2 above)
Comment (tchaton) [reply pl_examples/run_examples-args.sh:8]: @Borda Mind checking this PR again ?

File: .azure-pipelines/gpu-tests.yml [hunk 3]
```diff
@@ -104,10 +104,10 @@ jobs:
       displayName: 'Testing: benchmarks'
 
     - script: |
-        set -e
```
Comment (Borda) [inline .azure-pipelines/gpu-tests.yml:107]: you need to keep this, otherwise, it will pass with another bailing than the last bash

Comment (carmocca) [reply .azure-pipelines/gpu-tests.yml:107]: Done

**Review by Borda — APPROVED** [review]

**Review by SeanNaren — APPROVED** [review]

**Pull request merged**